lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -384,6 +384,14 @@ static bool loadIsMMAv3(Operation *loadOp) {
   if (!sharedEnc.getHasLeadingOffset())
     return false;
 
+  // In case LHS is in registers, don't pipeline for now TODO(ggengnv) is this necessary?
+  auto op = *alloc->getUsers().begin();
+  if (auto localLoad = dyn_cast<ttg::LocalLoadOp>(op)) {
+    auto resTy = cast<RankedTensorType>(localLoad->getResultTypes()[0]);
+    if (!resTy || isa<ttg::DotOperandEncodingAttr>(resTy.getEncoding()))
+      return false;
+  }
```

Comment:
this sounds like it will be inefficient? Why do we need that?

Reply:
this isn't necessary - I originally had pipelining logic enabled in this PR but there were concerns that this PR was getting too large, so I separated it into another [PR](https://github.com/openxla/triton/pull/19).

If you think it's more natural, I can add back the pipelining logic into this PR.

Reply:
Actually I just saw your other comment. I can keep the changes separate for now and combine them before merging.

Reply:
Regarding splitting the PR, I think the PR could be split between the changes to `kWidth`, which are simple and benevolent, and the hoisting + pipelining logic, which seem a bit trickier. @ggengnv how does that sound?

Reply:
sounds good to me.

by changes to `kWidth`, you mean the lowering of shared to register copies for MMAv3, correct?

the changes should be able to be split cleanly. only thing is that it might be hard to test by itself?

Reply:
everything but the hoisting pass, yes.
When it comes to testing, you can add a couple  lit tests that exercise `kWidth != 4 / elemSize`

Reply:
but yeah, since these changes are orthogonal splitting it in two shouldn't be terribly difficult. Tag me as a reviewer once you put it up and I'll approve it. That should make this PR much more lean, and it should be then fine to merge this PR and the pipeline PR into a single manageable PR